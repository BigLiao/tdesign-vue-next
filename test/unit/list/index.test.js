import { mount } from '@vue/test-utils';
import { List } from '@/src/list/index.ts';

describe('List', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(List, {
      propsData: {
        header: 'header',
        footer: 'footer',
        asyncLoading: 'asyncLoading',
        size: 'large',
      },
    });
  });

  it('equals header to "header"', () => {
    expect(cmp.vm.header).toEqual('header');
  });
  it('equals footer to "footer"', () => {
    expect(cmp.vm.footer).toEqual('footer');
  });
  it('equals asyncLoading to "asyncLoading"', () => {
    expect(cmp.vm.asyncLoading).toEqual('asyncLoading');
  });
  it('equals size to "large"', () => {
    expect(cmp.vm.size).toEqual('large');
  });
  it('equals layout to "horizontal"', () => {
    expect(cmp.vm.layout).toEqual('horizontal');
  });
});
