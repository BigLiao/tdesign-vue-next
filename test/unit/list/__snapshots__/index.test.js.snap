// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List :props :asyncLoading is a function, () => VNode. 1`] = `
<div
  class="t-list t-size-m"
>
  
  
  <!---->
  <ul
    class="t-list-items"
  >
    <!---->
  </ul>
  <!---->
  
  <div
    class="t-list__load"
  >
    <p>
      This is a loading
    </p>
  </div>
  
</div>
`;

exports[`List :props :asyncLoading is a string 1`] = `
<div
  class="t-list t-size-m"
>
  
  
  <!---->
  <ul
    class="t-list-items"
  >
    
    text
    
  </ul>
  <!---->
  
  <div
    class="t-list__load t-list__load--loading"
  >
    <div>
      <svg
        class="t-icon t-icon-loading"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 1.5a6.5 6.5 0 000 13v-1.63A4.87 4.87 0 1112.88 8h1.62A6.5 6.5 0 008 1.5z"
          fill="currentColor"
          fillOpacity="0.9"
        />
      </svg>
      <span>
        正在加载中，请稍等
      </span>
    </div>
  </div>
  
</div>
`;

exports[`List :props :asyncLoading is load-more 1`] = `
<div
  class="t-list t-size-m"
>
  
  
  <!---->
  <ul
    class="t-list-items"
  >
    
    text
    
  </ul>
  <!---->
  
  <div
    class="t-list__load t-list__load--load-more"
  >
    <span>
      点击加载更多
    </span>
  </div>
  
</div>
`;

exports[`List :props :footer is a function, () => VNode. 1`] = `
<div
  class="t-list t-size-m"
>
  
  
  <!---->
  <ul
    class="t-list-items"
  >
    <!---->
  </ul>
  <div
    class="t-list__footer"
  >
    <p>
      This is a footer
    </p>
  </div>
  
  <div
    class="t-list__load"
  >
    <!---->
  </div>
  
</div>
`;

exports[`List :props :footer is a string 1`] = `
<div
  class="t-list t-size-m"
>
  
  
  <!---->
  <ul
    class="t-list-items"
  >
    
    text
    
  </ul>
  <div
    class="t-list__footer"
  >
    footer
  </div>
  
  <div
    class="t-list__load"
  >
    <!---->
  </div>
  
</div>
`;

exports[`List :props :header is a function, () => VNode. 1`] = `
<div
  class="t-list t-size-m"
>
  
  
  <div
    class="t-list__header"
  >
    <p>
      This is a header
    </p>
  </div>
  <ul
    class="t-list-items"
  >
    <!---->
  </ul>
  <!---->
  
  <div
    class="t-list__load"
  >
    <!---->
  </div>
  
</div>
`;

exports[`List :props :header is a string 1`] = `
<div
  class="t-list t-size-m"
>
  
  
  <div
    class="t-list__header"
  >
    header
  </div>
  <ul
    class="t-list-items"
  >
    
    text
    
  </ul>
  <!---->
  
  <div
    class="t-list__load"
  >
    <!---->
  </div>
  
</div>
`;

exports[`List :props :layout 1`] = `
<div
  class="t-list t-size-m"
>
  
  
  <!---->
  <ul
    class="t-list-items"
  >
    
    text
    
  </ul>
  <!---->
  
  <div
    class="t-list__load"
  >
    <!---->
  </div>
  
</div>
`;

exports[`List :props :size 1`] = `
<div
  class="t-list t-size-l"
>
  
  
  <!---->
  <ul
    class="t-list-items"
  >
    
    text
    
  </ul>
  <!---->
  
  <div
    class="t-list__load"
  >
    <!---->
  </div>
  
</div>
`;

exports[`List :props :split 1`] = `
<div
  class="t-list t-size-m t-list--split"
>
  
  
  <!---->
  <ul
    class="t-list-items"
  >
    
    text
    
  </ul>
  <!---->
  
  <div
    class="t-list__load"
  >
    <!---->
  </div>
  
</div>
`;

exports[`List :props :stripe 1`] = `
<div
  class="t-list t-size-m"
>
  
  
  <!---->
  <ul
    class="t-list-items"
  >
    
    text
    
  </ul>
  <!---->
  
  <div
    class="t-list__load"
  >
    <!---->
  </div>
  
</div>
`;

exports[`ListItem :props :action 1`] = `
<li
  action="action"
  class="t-list-item"
>
  <div
    class="t-list-item-main"
  >
    
    text
    
    <!---->
  </div>
</li>
`;

exports[`ListItem :props :content 1`] = `
<li
  class="t-list-item"
  content="content"
>
  <div
    class="t-list-item-main"
  >
    
    text
    
    <!---->
  </div>
</li>
`;

exports[`ListItem :props :content is a function, () => VNode. 1`] = `
<li
  class="t-list-item"
  content="() => (0, _vue.createVNode)(\\"p\\", null, [(0, _vue.createTextVNode)(\\"This is a content\\")])"
>
  <div
    class="t-list-item-main"
  >
    <!---->
    <!---->
  </div>
</li>
`;

exports[`ListItem :props :default 1`] = `
<li
  class="t-list-item"
  default="default"
>
  <div
    class="t-list-item-main"
  >
    
    text
    
    <!---->
  </div>
</li>
`;

exports[`ListItem :props :default is a function, () => VNode. 1`] = `
<li
  class="t-list-item"
  default="() => (0, _vue.createVNode)(\\"p\\", null, [(0, _vue.createTextVNode)(\\"default\\")])"
>
  <div
    class="t-list-item-main"
  >
    <!---->
    <!---->
  </div>
</li>
`;

exports[`ListItemMeta :props :description 1`] = `
<div
  class="t-list-item__meta"
>
  
  <!---->
  <div
    class="t-list-item__meta-content"
  >
    <!---->
    <p
      class="t-list-item__meta-description"
    >
      description
    </p>
  </div>
  
</div>
`;

exports[`ListItemMeta :props :description is a function, () => VNode. 1`] = `
<div
  class="t-list-item__meta"
>
  
  <!---->
  <div
    class="t-list-item__meta-content"
  >
    <!---->
    <p
      class="t-list-item__meta-description"
    >
      <p>
        description
      </p>
    </p>
  </div>
  
</div>
`;

exports[`ListItemMeta :props :image 1`] = `
<div
  class="t-list-item__meta"
>
  
  <div
    class="t-list-item__meta-avatar"
  >
    <img
      src="image"
    />
  </div>
  <div
    class="t-list-item__meta-content"
  >
    <!---->
    <!---->
  </div>
  
</div>
`;

exports[`ListItemMeta :props :title 1`] = `
<div
  class="t-list-item__meta"
>
  
  <!---->
  <div
    class="t-list-item__meta-content"
  >
    <h3
      class="t-list-item__meta-title"
    >
      title
    </h3>
    <!---->
  </div>
  
</div>
`;

exports[`ListItemMeta :props :title is a function, () => VNode. 1`] = `
<div
  class="t-list-item__meta"
>
  
  <!---->
  <div
    class="t-list-item__meta-content"
  >
    <h3
      class="t-list-item__meta-title"
    >
      <p>
        title
      </p>
    </h3>
    <!---->
  </div>
  
</div>
`;
